main.cpp

```diff
@@ -7,35 +7,49 @@
 #include <clang/ASTMatchers/ASTMatchFinder.h>
 #include <clang/ASTMatchers/ASTMatchers.h>
 #include <clang/Basic/Diagnostic.h>
+#include <clang/Frontend/CompilerInstance.h>
 #include <clang/Frontend/FrontendActions.h>
+#include <clang/Rewrite/Core/Rewriter.h>
 #include <clang/Tooling/CommonOptionsParser.h>
 #include <clang/Tooling/Tooling.h>
-#include <clang/Rewrite/Core/Rewriter.h>
-#include <clang/Frontend/CompilerInstance.h>
 
 using namespace clang;
 using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+    Rewriter& mRewriter;
+
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
+    CastCallBack(Rewriter& rewriter) : mRewriter(rewriter) {
+
     };
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    void run(const MatchFinder::MatchResult& Result) override {
+        auto castNode = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast"); // Get AST node bind to id = "cast"
+
+        if (castNode) {
+            auto leftSL = castNode->getLParenLoc(); // Get left SourceLocation
+            auto rightSL = castNode->getRParenLoc(); // Get right SourceLocation
+
+            // Initial source = "(int)f"
+
+            mRewriter.ReplaceText(leftSL, 1, "static_cast<"); // Remove '(' --> "int)f" and insert "static_cast<" --> "static_cast<int)f"
+            mRewriter.ReplaceText(rightSL, 1, ">("); // Remove ')' --> "static_cast<intf" and insert ">(" --> "static_cast<int>(f"
+            mRewriter.InsertText(Lexer::getLocForEndOfToken(castNode->getEndLoc(), 0, *Result.SourceManager, Result.Context->getLangOpts()), ")"); // Insert ")" behind "f" --> "static_cast<int>(f)"

```

Comment:
Please describe the reason of using the`Lexer::getLocForEndOfToken()` function here.
What would be the difference if `rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")");` was used here?
What would be the difference if  `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")");` was used here?

Reply:
I was looking for a function to help jump to the end of the line where the castNode is located, that is, before ";".
It's generally not entirely correct to use getLocForEndOfToken() to accomplish this task, but at that time I was not aware of this.
If you use rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")"); then ")" will be inserted immediately after styleCastExpr + 1(offset), that is, if the length of the variable name is more than 1 character, this will not give us the desired result.
If we use rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")"); then we will insert after the token ")", that is, we will insert after the name of the variable what we need.
Now I can say it's better to use rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")"); to solve the problem.

Reply:
Thanks for the answer, accepted!
Both options `Lexer::getLocForEndOfToken(styleCastExpr->getEndLoc(),...)` and `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(),...)` can be applied here, they give the same expected result.

Reply:
`It's generally not entirely correct to use getLocForEndOfToken() to accomplish this task` this is not correct, `getLocForEndOfToken()` can be used to accomplish this task. Sorry, I missed this suggestion earlier.